scheduler/services/chronicle/__init__.py

```diff
@@ -1,36 +1,65 @@
 import re
 import os
-from typing import Dict, Set, FrozenSet
-from datetime import date, datetime, timedelta
+from io import BytesIO
+from typing import Dict, FrozenSet, Optional, List
+from datetime import date, datetime, timedelta, time
 from dataclasses import dataclass
+from typing import final, Union
 
-from astropy.time import Time
-from lucupy.sky import night_events
 from lucupy.minimodel import ALL_SITES, Site
 
 from definitions import ROOT_DIR
 from scheduler.services.abstract import ExternalService
 
 
-@dataclass(frozen=True)
+@final
+@dataclass
+class TwilightTime:
+    value: str = "twi"
+
+    def __sub__(self, other):
+        # When an unresolved twilight is calculated
+        # show a huge number to indentify error.
+        return 9999
+
+
+@dataclass
 class Interruption:
     """
     Parent class for any interruption in the night that would
     cause missing time of observation.
     """
-    start: datetime
-    time_loss: timedelta
+    start: Union[datetime, TwilightTime]
     reason: str
+    id: Optional[str] = None  # For FR only
```

Comment:
Can't inherit from a dataclass which has a default parameter. It would try to put the init as:
```
def __init__(param1, param2=None, param3, ... etc)
```